### PR TITLE
fix: support Elixir 1.17

### PR DESCRIPTION
**Pull request by joshuap**
https://hexdocs.pm/elixir/1.17.3/compatibility-and-deprecations.html

- [ ] Needs changelog entry

### CODE REVIEW

**Comment by joshuap** [comment]
Regarding https://github.com/honeybadger-io/honeybadger-elixir/pull/575/commits/038ee2200093b75c6b9a630b533a37bb809d1cdb, here is the diff between the Elixir 1.15 and 1.17 events that are being passed to `handle_event/2` as a part of the Logger `message` object:

https://github.com/honeybadger-io/honeybadger-elixir/blob/bca38d914eb319a77838305fb4d54fbf02271823/lib/honeybadger/logger.ex#L27

https://github.com/honeybadger-io/honeybadger-elixir/blob/bca38d914eb319a77838305fb4d54fbf02271823/lib/honeybadger/logger.ex#L30

```diff
GenServer extract_details args: [
  [
    "GenServer ",
    "MyGenServer",
    " terminating",
    [
      [10 | "** (KeyError) key :bad_key not found in: %{}"],
      ["\n    " | ":erlang.map_get(:bad_key, %{})"],
      ["\n    " |
       "test/honeybadger/logger_test.exs:25: Honeybadger.LoggerTest.MyGenServer.handle_cast/2"],
      ["\n    " |
       "(stdlib 6.1.1) gen_server.erl:2371: :gen_server.try_handle_cast/3"],
      ["\n    " |
       "(stdlib 6.1.1) gen_server.erl:2433: :gen_server.handle_msg/6"],
      ["\n    " |
       "(stdlib 6.1.1) proc_lib.erl:329: :proc_lib.init_p_do_apply/3"]
    ],
+    [],
    "\nLast message",
    [],
    ": ",
    "{:\"$gen_cast\", :raise_error}"
  ],
  "\nState: ",
  "%{}"
]
```

```diff
:gen_event handler extract_details args: [
  [
    ":gen_event handler ",
    "Honeybadger.LoggerTest.MyEventHandler",
    " installed in ",
    "#PID<0.7227.0>",
    " terminating",
    [
      [10 | "** (RuntimeError) Oops"],
      ["\n    " |
       "test/honeybadger/logger_test.exs:58: Honeybadger.LoggerTest.MyEventHandler.handle_event/2"],
      ["\n    " |
       "(stdlib 6.1.1) gen_event.erl:1921: :gen_event.server_update/4"],
      ["\n    " |
       "(stdlib 6.1.1) gen_event.erl:1903: :gen_event.server_notify/4"],
      ["\n    " | "(stdlib 6.1.1) gen_event.erl:1640: :gen_event.handle_msg/6"],
      ["\n    " |
       "(stdlib 6.1.1) proc_lib.erl:329: :proc_lib.init_p_do_apply/3"]
    ],
+    [],
    "\nLast message: ",
    ":raise_error"
  ],
  "\nState: ",
  "{}"
]
```

**Comment by joshuap** [comment]
Hey @sorentwo! Are you around and is there any chance you could give this a quick look?

File: .github/workflows/elixir.yml [hunk 1]
```diff
@@ -18,11 +18,14 @@ jobs:
       matrix:
         include:
           - pair:
-              elixir: '1.11'
-              otp: '21.3'
+              elixir: '1.14'
+              otp: '26.0'
```
Comment (sorentwo) [inline .github/workflows/elixir.yml:22]: The minimum supported OTP version is 23.0 for Elixir v1.14. The strategy here was to test with the oldest and newest versions.

```suggestion
              otp: '23.3'
```

**Review by sorentwo — APPROVED** [review]
One minor suggestion to maximize the test matrix, but this looks good otherwise. Note that Elixir v1.18 will also be released soon (it's in an RC currently).

**Comment by joshuap** [comment]
Thanks!

**Pull request merged**